db/migrate/20140615175553_init_schema.rb

```diff
@@ -0,0 +1,871 @@
+class InitSchema < ActiveRecord::Migration
+  def down
+    raise 'Can not revert initial migration'
+  end
+
+  def up
+
+    create_table 'alternative_iris', force: true do |t|
+      t.text     'iri'
+      t.integer  'ontology_id', null: false
+      t.datetime 'created_at',  null: false
+      t.datetime 'updated_at',  null: false
+    end
+
+    add_index 'alternative_iris', ['ontology_id', 'iri'], name: 'index_alternative_iris_on_ontology_id_and_iri', unique: true
+
+    create_table 'basic_proofs', force: true do |t|
+      t.string   'prover'
+      t.text     'proof'
+      t.integer  'logic_mapping_id', null: false
+      t.integer  'sentence_id',      null: false
+      t.datetime 'created_at',       null: false
+      t.datetime 'updated_at',       null: false
+    end
+
+    add_index 'basic_proofs', ['logic_mapping_id'], name: 'index_basic_proofs_on_logic_mapping_id'
+    add_index 'basic_proofs', ['sentence_id'], name: 'index_basic_proofs_on_sentence_id'
+
+    create_table 'c_edges', force: true do |t|
+      t.integer 'parent_id', null: false
+      t.integer 'child_id',  null: false
+    end
+
+    add_index 'c_edges', ['parent_id', 'child_id'], name: 'index_c_edges_on_parent_id_and_child_id', unique: true
+
+    create_table 'categories', force: true do |t|
+      t.text     'name',       null: false
+      t.string   'ordinal'
+      t.datetime 'created_at', null: false
+      t.datetime 'updated_at', null: false
+    end
+
+    create_table 'categories_ontologies', force: true do |t|
+      t.integer 'ontology_id', null: false
+      t.integer 'category_id', null: false
+    end
+
+    add_index 'categories_ontologies', ['category_id', 'ontology_id'], name: 'index_c_vertices_ontologies_on_c_vertex_id_and_ontology_id', unique: true
+    add_index 'categories_ontologies', ['ontology_id', 'category_id'], name: 'index_c_vertices_ontologies_on_ontology_id_and_c_vertex_id'
+
+    create_table 'code_references', force: true do |t|
+      t.integer  'begin_line'
+      t.integer  'end_line'
+      t.integer  'begin_column'
+      t.integer  'end_column'
+      t.integer  'referencee_id'
+      t.string   'referencee_type'
+      t.datetime 'created_at',      null: false
+      t.datetime 'updated_at',      null: false
+    end
+
+    create_table 'comments', force: true do |t|
+      t.integer  'commentable_id',   null: false
+      t.string   'commentable_type', null: false
+      t.integer  'user_id',          null: false
+      t.text     'text',             null: false
+      t.datetime 'created_at',       null: false
+      t.datetime 'updated_at',       null: false
+    end
+
+    add_index 'comments', ['commentable_id', 'commentable_type', 'id'], name: 'index_comments_on_commentable_and_id'
```

Comment:
Metrics/LineLength: Line is too long. [116/80]<br>Style/WordArray: Use `%w` or `%W` for array of words.